dusty/commands/env.py

```diff
@@ -0,0 +1,94 @@
+import functools
+import os
+
+from prettytable import PrettyTable
+
+from ..log import log_to_client
+from ..compiler.spec_assembler import get_assembled_specs, get_specs
+from ..config import get_env_config, save_config_value
+from .. import constants
+from ..payload import daemon_command
+
+def _save_env(global_env_dict):
+    save_config_value(constants.CONFIG_ENV_KEY, global_env_dict)
+
+def _assert_app_or_service_exists(app_or_service):
+    """
+    get_app_or_service will throw a KeyError if the app_or_service
+    is not found.
+    """
+    get_specs().get_app_or_service(app_or_service)
+
+@daemon_command
+def list_app_or_service(app_or_service):
+    _assert_app_or_service_exists(app_or_service)
+    env_dict = get_env_config().get(app_or_service)
+    if not env_dict:
+        log_to_client('No environment overrides specified for {}'.format(app_or_service))
+        return
+    table = PrettyTable(["Variable", "Value"])
+    for var_name, value in env_dict.iteritems():
+        table.add_row([var_name, value])
+    log_to_client(table.get_string())
+
+@daemon_command
+def list_all():
+    global_env_dict = get_env_config()
+    if not global_env_dict:
+        log_to_client('No environment overrides specified')
+        return
+    active_containers = set([cont.name for cont in get_assembled_specs().get_apps_and_services()])
+    log_to_client('Listing environment overrides for activated apps and services:')
+    table = PrettyTable(["App or Service", "Variable", "Value"])
+    for app_or_service, env_dict in global_env_dict.iteritems():
+        if app_or_service in active_containers:
+            for var_name, value in env_dict.iteritems():
+                table.add_row([app_or_service, var_name, value])
+    log_to_client(table.get_string())
+
+@daemon_command
+def set_from_file(app_or_service, local_file_path):
+    _assert_app_or_service_exists(app_or_service)
+    if not os.path.exists(local_file_path):
```

Comment:
This should also check whether the file is readable, I think you can use `os.stat` for that